evap/rewards/forms.py

```diff
@@ -1,14 +1,95 @@
+from datetime import date
+
 from django import forms
+from django.core.exceptions import ValidationError
+from django.core.validators import MaxValueValidator, StepValueValidator
+from django.db import transaction
+from django.utils.translation import gettext as _
 
-from evap.rewards.models import RewardPointRedemptionEvent
+from evap.rewards.models import RewardPointRedemption, RewardPointRedemptionEvent
+from evap.rewards.tools import reward_points_of_user
 
 
 class RewardPointRedemptionEventForm(forms.ModelForm):
     class Meta:
         model = RewardPointRedemptionEvent
-        fields = ("name", "date", "redeem_end_date")
+        fields = ("name", "date", "redeem_end_date", "step")
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.fields["date"].localize = True
         self.fields["redeem_end_date"].localize = True
+
+
+class RedeemRewardPointsForm(forms.Form):
+    event = forms.ModelChoiceField(queryset=RewardPointRedemptionEvent.objects.all(), widget=forms.HiddenInput())
+    points = forms.IntegerField(min_value=0, label="")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if not self.initial:
+            return
+
+        self.fields["points"].validators.append(MaxValueValidator(self.initial["total_points_available"]))
+        self.fields["points"].widget.attrs["max"] = self.initial["total_points_available"]
+
+        self.fields["points"].validators.append(StepValueValidator(self.initial["event"].step))
+        self.fields["points"].widget.attrs["step"] = self.initial["event"].step
+
+        self.fields["points"].help_text = _("multiples of {}").format(self.initial["event"].step)
+
+    def clean(self):
```

Comment:
as discussed, this can probably be removed because both error cases can't occur here